actors/common/src/main/java/org/sunbird/learner/util/Util.java

```diff
@@ -1463,24 +1467,86 @@ public static void checkProfileCompleteness(Map<String, Object> userMap) {
     ProfileCompletenessService profileService = ProfileCompletenessFactory.getInstance();
     Map<String, Object> profileResponse = profileService.computeProfile(userMap);
     userMap.putAll(profileResponse);
-    Map<String, String> profileVisibility =
+  }
+
+  public static void checkUserProfileVisibility(Map<String, Object> userMap, ActorRef actorRef) {
+
+    Map<String, String> userProfileVisibilityMap =
         (Map<String, String>) userMap.get(JsonKey.PROFILE_VISIBILITY);
-    if (null != profileVisibility && !profileVisibility.isEmpty()) {
-      Map<String, Object> profileVisibilityMap = new HashMap<>();
-      for (String field : profileVisibility.keySet()) {
-        profileVisibilityMap.put(field, userMap.get(field));
+    Map<String, String> completeProfileVisibilityMap =
+        getCompleteProfileVisibilityMap(userProfileVisibilityMap, actorRef);
+
+    if (null != completeProfileVisibilityMap && !completeProfileVisibilityMap.isEmpty()) {
+      Map<String, Object> privateFieldsMap = new HashMap<>();
+      for (String field : completeProfileVisibilityMap.keySet()) {
+        if (JsonKey.PRIVATE.equals(completeProfileVisibilityMap.get(field))) {
+          privateFieldsMap.put(field, userMap.get(field));
+        }
       }
       ElasticSearchUtil.upsertData(
           ProjectUtil.EsIndex.sunbird.getIndexName(),
           ProjectUtil.EsType.userprofilevisibility.getTypeName(),
           (String) userMap.get(JsonKey.USER_ID),
-          profileVisibilityMap);
-      UserUtility.updateProfileVisibilityFields(profileVisibilityMap, userMap);
+          privateFieldsMap);
+      UserUtility.updateProfileVisibilityFields(privateFieldsMap, userMap);
     } else {
       userMap.put(JsonKey.PROFILE_VISIBILITY, new HashMap<String, String>());
     }
   }
 
+  public static Map<String, String> getCompleteProfileVisibilityPrivateMap(
+      Map<String, String> userProfileVisibilityMap, ActorRef actorRef) {
+    Map<String, String> completeProfileVisibilityMap =
+        getCompleteProfileVisibilityMap(userProfileVisibilityMap, actorRef);
+    for (String key : completeProfileVisibilityMap.keySet()) {
+      if (JsonKey.PUBLIC.equals(completeProfileVisibilityMap.get(key))) {
+        completeProfileVisibilityMap.remove(key);
+      }
+    }
+    return completeProfileVisibilityMap;
+  }
+
+  public static Map<String, String> getCompleteProfileVisibilityMap(
+      Map<String, String> userProfileVisibilityMap, ActorRef actorRef) {
+    String defaultProfileVisibility =
+        ProjectUtil.getConfigValue(JsonKey.SUNBIRD_DEFAULT_USER_PROFILE_VISIBILITY);
+    if (!(defaultProfileVisibility.equals(JsonKey.PUBLIC)
```

Comment:
Use ignore case in comparison.

Use JsonKey.PUBLIC.equals as that would avoid null pointer issue.